src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -114,10 +116,22 @@ private enum Flags : ulong
             FragmentIriCanonical = 0x40000000000,
             IriCanonical = 0x78000000000,
             UnixPath = 0x100000000000,
+
+            CustomParser_ParseMinimalAlreadyCalled = 1UL << 62,
+
+            Debug_LeftConstructor = 1UL << 63
         }
 
-        private Flags _flags;
-        private UriInfo _info = null!; // initialized during ctor via helper
+        [Conditional("DEBUG")]
+        private void DebugSetLeftCtor()
+        {
+            _flags |= Flags.Debug_LeftConstructor;
+        }
+        [Conditional("DEBUG")]
+        internal void DebugAssertInCtor()
+        {
+            Debug.Assert((_flags & Flags.Debug_LeftConstructor) == 0);
+        }
```

Comment:
Uri has some methods that update the flags directly, but that is okay as they are called before exiting the ctor or under a lock. Other methods make flag changes on a local Flags copy and only _save_ the result in a synchronized fashion.

This allows us to assert that behavior and make sure that methods modifying flags directly are only called from ctors.